File: DataSummitSolution/DataSummitUI/src/app/projects/models/project.ts
export class Project {

	ProjectId: number;
	Name: string;
	CompanyId: number;
	CreatedDate: Date | string;

	constructor(projectId?: number, 
		name?: string, 
		companyId?: number,
		createdDate?: Date | string)
	{
		this.ProjectId = projectId || 0;
		this.Name = name || "";
		this.CompanyId = companyId || 0;
		this.CreatedDate = createdDate || new Date(Date.now());
	}
}

File: DataSummitSolution/DataSummitWeb/Models/Project.cs
using DataSummitHelper.Interfaces;
using System;

namespace DataSummitWeb.Models
{
    /// <summary>
    /// </summary>
    public sealed class Project
    {
        public int ProjectId { get; set; }
        public string Name { get; set; }
        public int CompanyId { get; set; }
        public DateTime CreatedDate { get; set; }

        public static Project FromDto(ProjectDto dto) => new Project
        {
            ProjectId = dto.ProjectId,
            Name = dto.ProjectName,
            CompanyId = dto.CompanyId,
            CreatedDate = dto.CreatedDate
        };

        public ProjectDto ToDto() => new ProjectDto()
        {
            ProjectId = ProjectId,
            ProjectName = Name,
            CompanyId = CompanyId,
            CreatedDate = CreatedDate
        };
    }
}
